### PR TITLE
Feat/api calls

### PR DESCRIPTION
**Pull request by colefiscus**
## What does this PR do?

This PR builds the API call functions inside of util.js and utilizes those functions inside of App. Additionally this PR adds styling for the page and refactors a lot of the code to be able to use the API data instead of the model data we used to initially build the page. And finally, this PR adds error handling with the API call functions as well as within App.js

## What does it fix (if anything)?

We have "fixed" the code to use the API data instead of the model data. This mostly entailed properly passing down props and adjusting how we were accessing that data.

## Is this a feature or a fix?

Feature & fix.

## Where should the reviewer start?

The reviewer should start by looking at the functions inside of util.js, then look at where those functions are being called inside of App.js and follow the flow of data to our other components to see how the project is working.

Also noteworthy in this PR is the added error handling within the API call functions as well as inside of App.js.

## What comes next after this is merged?

Next up will be incorporating Router into our project, increasing the styling of the pages, attempt to add the movie's trailers into the movie details component.

### CODE REVIEW

**Comment by colefiscus** [comment]
@sertmer @letakeane

File: src/Components/App/App.js [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
+        this.setState({ error: "Oops! Something went wrong!" })
+      })
+  }
+
   handleClick = (id) => {
-    const current = this.state.movies.find(movie => movie.id === id)
-    this.setState({currentMovie: current})
+    const current = this.state.movies.movies.find(movie => movie.id === id)
+    getSingleMovie(current.id)
+      .then(movie => this.setState({currentMovie: movie}))
```
Comment (rachelbuchta) [inline src/Components/App/App.js:35]: I'm thinking we need a .catch here too

File: src/util.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+export const getAllMovies = () => {
+return fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')  
+  .then(response => { 
+      console.log(response.status)                  
+      if(!response.ok) {
```
Comment (rachelbuchta) [inline src/util.js:5]: I'm not sure if pinpointing a 500 status code in this way is absolutely necessary, but it works for now

File: .eslintcache [hunk 3]
```diff
@@ -1 +1 @@
...
...Tomatillos/src/index.js",[],["35","36"],"/Users/rachelbuchta/turing/mod3/projects/rancid-tomatillos/rancidTomatillos/src/Components/App/App.js",["37"],"/Users/rachelbuchta/turing/mod3/projects/rancid-tomatillos/rancidTomatillos/src/util.js",[],"/Users/rachelbuchta/turing/mod3/projects/rancid-tomatillos/rancidTomatillos/src/Components/Header/Header.js",[],"/Users/rachelbuchta/turing/mod3/projects/rancid-tomatillos/rancidTomatillos/src/Components/Footer/Footer.js",[],"/Users/rachelbuchta/turing/mod3/projects/rancid-tomatillos/rancidTomatillos/src/Components/Movies/Movies.js",[],"/Users/rachelbuchta/turing/mod3/projects/rancid-tomatillos/rancidTomatillos/src/Components/MovieDetails/MovieDetails.js",[],"/Users/rachelbuchta/turing/mod3/projects/rancid-tomatillos/rancidTomatillos/src/Components/Movie/Movie.js",[],{"ruleId":"38","replacedBy":"39"},{"ruleId":"40","replacedBy":"41"},{"ruleId":"42","severity":1,"message":"43","line":7,"column":40,"nodeType":"44","messageId":"45","endLine":7,"endColumn":59},"no-native-reassign",["46"],"no-negated-in-lhs",["47"],"no-unused-vars","'getSingleMovieVideo' is defined but never used.","Identifier","unusedVar","no-global-assign","no-unsafe-negation"]
```
Comment (sertmer) [inline .eslintcache:1]: Might be good to add this file to your gitignore

File: src/Components/App/App.js [hunk 4]
```diff
@@ -1,23 +1,38 @@
...
+import { getAllMovies, getSingleMovie, getSingleMovieVideo } from '../../util'
 
 class App extends Component {
   constructor() {
     super()
     this.state = { 
-      movies: movieData.movies,
-      currentMovie: ''
+      movies: [],
+      currentMovie: '',
```
Comment (sertmer) [inline src/Components/App/App.js:14]: currentMovie will eventually be an object - any reason it starts as a string?

File: src/Components/App/App.js (hunk 1 above)
Comment (sertmer) [reply src/Components/App/App.js:35]: I'd agree with that!

File: src/Components/App/App.js [hunk 5]
```diff
@@ -29,12 +44,19 @@ class App extends Component {
...
-        {!this.state.currentMovie && (
+        {this.state.isLoading && !this.state.error &&
+        ( <h2 className="userMsg">Loading...</h2> )}
+
+        {this.state.error && (
+          <h2 className="userMsg">{this.state.error}</h2>
+        )}
+
+        {!this.state.isLoading && !this.state.currentMovie && (
         <Movies movies={this.state.movies} handleClick={this.handleClick}/>
```
Comment (sertmer) [inline src/Components/App/App.js:55]: lots of this.state here - maybe we could destructure isLoading, error, currentmovie etc to save us some keystrokes?

File: src/util.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+export const getAllMovies = () => {
+return fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')  
+  .then(response => { 
+      console.log(response.status)
```
Comment (sertmer) [inline src/util.js:4]: don't forget to nuke these console.logs before pushing up

File: src/util.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+export const getAllMovies = () => {
+return fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')  
+  .then(response => { 
+      console.log(response.status)                  
+      if(!response.ok) {
+        throw new Error('Something went wrong.');
+      } else if (response.status === 500) {
```
Comment (sertmer) [inline src/util.js:7]: yeah, what if it's a 404? maybe we could have a conditional check if it's in the 400 range, or if it's great that 499, do this...

**Pull request merged**